server/src/main/java/org/jboss/as/server/deployment/DeploymentOverlayDeploymentUnitProcessor.java

```diff
@@ -65,6 +66,8 @@
 
     private final ContentRepository contentRepository;
 
+    private static final String WAR_EXTENSION = ".war";
```

Comment:
We should not be relying on WAR_EXTENSION. Instead I think this should just use org.jboss.as.server.deployment.MountExplodedMarker to check if it is a deployment that is mounted exploded.